CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/CollectionRequestHandler.swift

```diff
@@ -0,0 +1,142 @@
+//RequestHandler.swift
+//CBLTestServer-iOS
+
+import Foundation
+import CouchbaseLiteSwift
+
+public class CollectionRequestHandler {
+    public static let VOID: String? = nil
+    fileprivate var _pushPullReplListener:NSObjectProtocol?
+    
+    public func handleRequest(method: String, args: Args) throws -> Any? {
+        switch method {
+        case "collection_defaultCollection":
+            let database: Database = (args.get(name:"database"))!
+            let collection: Collection = try (database.defaultCollection())!
+            return collection
+            
+        case "collection_getCollectionName":
+            let collection: Collection = (args.get(name:"collection"))!
+            return collection.name
+            
+        case "collection_createCollection":
+            let scopeName: String = (args.get(name: "scopeName"))!
+            let collectionName: String = (args.get(name: "collectionName"))!
+            let database: Database = args.get(name: "database")!
+            let collection: Collection = try database.createCollection(name: collectionName, scope: scopeName)
+            return collection
+            
+        case "collection_collectionNames":
+            let scope: String = (args.get(name:"scopeName"))!
+            let database: Database = (args.get(name:"database"))!
+            var names = [String]()
+            let collectionNames = try database.collections(scope: scope)
+            for collection in collectionNames{
+                let collectionObject: Collection = collection
+                names.append(collectionObject.name)
+            }
+            return names
+        
+        case "collection_deleteCollection":
+            let scopeName: String = (args.get(name:"scopeName"))!
+            let database: Database = (args.get(name:"database"))!
+            let collectionName: String = (args.get(name:"collectionName"))!
+            do{
+                try database.deleteCollection(name: collectionName, scope: scopeName)
+            } catch{
+                return error
```

Comment:
if this try/catch is added in case of not returning result, then if an error happens, you might want to do something other than return.